spec/unit/middleware/rewrite_spec.rb

```diff
@@ -26,8 +27,25 @@ def app
   context "server is run with job_id 42" do
     let(:job_id) { 42 }
     it 'responds with 422 on job_id mismatch' do
-      response = get("/jobs/666/uptime")
+      response = get "/api/v1/jobs/666/uptime"
       expect(response.status).to eq(422)
     end
   end
+
+  describe '/machines/logs/message' do
+    it 'rewrites logs route' do
+      expect(reporter).to receive(:send_log)
+      response = post "/api/v1/machines/logs/message",
+        { messageText: "foo" },
+        'x-MachineId' => 123
+      expect(last_request.form_data?).to be true
```

Comment:
please add also expectation on reporter

     expect(reporter).to receive(:send_log).with...